File: CumulativeSum/source.py
n, q = map(int, input().split())
a = list(map(int, input().split()))

# 総来場者数の配列を0で初期化
sum = [0] * (n+1)
for i in range(1, n+1):
    sum[i] = a[i-1] + sum[i-1]

# l日目からr日目までの総来場者数を計算
answer = []
for _ in range(q):
    left, right = map(int, input().split())
    result = sum[right] - sum[left-1]
    answer.append(result)

# 回答の出力
for x in answer:
    print(x)

File: CumulativeSum/source.cpp
#include <iostream>
#include <vector>

int main() {
    int n, q;
    std::cin >> n >> q;
    std::vector<int> a(n);
    for (int i = 0; i < n; ++i) {
        std::cin >> a[i];
    }

    // この実装だと実行制限時間に間に合わないため,計算量を減らした実装に変更してください
    for (int i = 0; i < q; ++i) {
        int left, right;
        std::cin >> left >> right;

        int answer = 0;
        for (int j = left - 1; j < right; ++j) {
            answer += a[j];
        }
        std::cout << answer << std::endl;
    }

    return 0;
}